Source/Tools/FEXLoader/LinuxSyscalls/Syscalls/Thread.cpp

```diff
@@ -187,11 +188,18 @@ namespace FEX::HLE {
   uint64_t ForkGuest(FEXCore::Core::InternalThreadState *Thread, FEXCore::Core::CpuStateFrame *Frame, uint32_t flags, void *stack, size_t StackSize, pid_t *parent_tid, pid_t *child_tid, void *tls) {
     // Just before we fork, we lock all syscall mutexes so that both processes will end up with a locked mutex
     FEX::HLE::_SyscallHandler->LockBeforeFork();
-    
+
+    const bool IsVFork = flags & CLONE_VFORK;
     pid_t Result{};
-    if (flags & CLONE_VFORK) {
-      // XXX: We don't currently support a vfork as it causes problems.
-      // Currently behaves like a fork, which isn't correct. Need to find where the problem is
+    int VForkFDs[2];
+    if (IsVFork) {
+      // Use pipes as a mechanism for knowing when the child process is exiting.
+      // FEX can't use `waitpid` for this since the child process may want to use it.
```

Comment:
Can't we use `WNOWAIT` to handle this?

Reply:
Sadly no.
```
wait4(488421, 0xffff94ece7a4, WNOWAIT, NULL) = -1 EINVAL (Invalid argument)
```
(wait4 implements waitpid on AArch64).
No way to work around this as far as I can tell.

Reply:
Oh, actually. `waitid(P_PID, pid, &Info, WNOWAIT | WEXITED);` Gets close. Except it doesn't return until exit.
We need to block until process exit OR execve. This would wait until the execve application exits which isn't the same semantics.